pkg/tarball/tarball.go

```diff
@@ -83,3 +85,76 @@ func DecodeTarball(reader io.Reader, baseDir string) error {
 
 	return nil
 }
+
+// TarDir tars up an entire directory and outputs the tarball to the specified output path.
+// If useGzip is true, it also gzips the resulting tarball.
+func TarDir(dir, outpath string, useGzip bool) error {
+	// ensure the src actually exists before trying to tar it
+	if _, err := os.Stat(dir); err != nil {
+		return errors.Wrapf(err, "tar unable to stat directory %v", dir)
+	}
+
+	outfile, err := os.Create(outpath)
+	if err != nil {
```

Comment:
Should we check for the type of err (if path already exists, for instance,  maybe just overwrite).

Reply:
Its on the server and is based on a UUID of the run; there really isn't a reasonable way that this would error now besides a crazy os issue. I don't think extra work on that path is necessary.